### PR TITLE
[VRD-193] When creating RMV from ER, we should reuse the ER's artifact's paths

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

**Review by conradoverta — CHANGES_REQUESTED** [review]
I'm very confused why we're adding this whole layer of parsing and then potentially re-creating the path. This makes the code very hard to read by embedding logic from a top level command (create from ER) all the way down to the artifact handler. From reading the code, I'd have no clue what this condition is trying to solve and I'm even having a hard time telling if the code is correct by reading it.

**Review by conradoverta — COMMENTED** [review]
Also, could we at least have a test for this given how critical this is?

**Comment by anandjakhaniya** [comment]
> I'm very confused why we're adding this whole layer of parsing and then potentially re-creating the path. This makes the code very hard to read by embedding logic from a top level command (create from ER) all the way down to the artifact handler. From reading the code, I'd have no clue what this condition is trying to solve and I'm even having a hard time telling if the code is correct by reading it.

We have multiple place where we are creating artifact and for all possibilities the last point is CommonArtifactHandler so i have handler path logic there.

- Logic will check about the path started with entity name as prefix 
- If it is there then we are considered path has new logic whatever entity is created that path. let say if path created in ER then it was started with ER entity name. we don't care about it will populated in MV or not because that common logic check for all entities at single place.
- If path does not started with entity name as prefix then we are creating new path with new logic in that entity.

I don't think there will be good place to add this logic and if we will try to simplify that logic for readability then we have to add that same logic multiple times at many places so i think it will create more errors instead of handling this logic at single place.

File: backend/common/src/main/java/ai/verta/modeldb/common/subtypes/CommonArtifactHandler.java [hunk 1]
```diff
@@ -130,18 +134,17 @@ protected abstract Query buildGetArtifactsQuery(
...
-      var path = validPrefix + "/" + entityId + "/" + artifact.getKey();
-      var storeTypePath =
-          !artifact.getPathOnly() ? artifactStoreConfig.storeTypePathPrefix() + path : "";
+      var path = artifact.getPath();
+      if (!isPopulateFromOtherEntity) {
+        var validPrefix = artifactStoreConfig.getPathPrefixWithSeparator() + this.entityName;
+        path = validPrefix + "/" + entityId + "/" + artifact.getKey();
+        artifact = artifact.toBuilder().setPath(path).build();
+      }
+      var storeTypePath = !artifact.getPathOnly() ? path : "";
```
Comment (conradoverta) [inline backend/common/src/main/java/ai/verta/modeldb/common/subtypes/CommonArtifactHandler.java:149]: `storeTypePath` here has a different value from before, so it has to be checked with care (we can't rely on previously working tests)

Comment (anandjakhaniya) [reply backend/common/src/main/java/ai/verta/modeldb/common/subtypes/CommonArtifactHandler.java:149]: no, it has same value. Before changes we are setting it if path_only is false other wise we will leave it empty. so here after changes we keep it as it is. it is not populate in artifact proto. it is populate in DB only.

**Review by conradoverta — APPROVED** [review]

File: backend/common/src/main/java/ai/verta/modeldb/common/subtypes/CommonArtifactHandler.java [hunk 2]
```diff
@@ -130,18 +134,23 @@ protected abstract Query buildGetArtifactsQuery(
...
+      var path = artifact.getPath();
+      if (!isPopulateFromOtherEntity) {
+        var validPrefix = artifactStoreConfig.getPathPrefixWithSeparator() + this.entityName;
+        path = validPrefix + "/" + entityId + "/" + artifact.getKey();
 
-      if (overwrite && isExists(entityId, artifact.getKey(), handle)) {
-        if (!artifact.getPath().startsWith(validPrefix)) {
-          artifact = artifact.toBuilder().setPath(path).build();
+        var filenameExtension = artifact.getFilenameExtension();
+        if (!filenameExtension.isEmpty() && !filenameExtension.endsWith("." + filenameExtension)) {
```
Comment (liuverta) [inline backend/common/src/main/java/ai/verta/modeldb/common/subtypes/CommonArtifactHandler.java]:
```suggestion
        if (!filenameExtension.isEmpty() && !path.endsWith("." + filenameExtension)) {
```

File: backend/src/main/java/ai/verta/modeldb/entities/ArtifactEntity.java [hunk 3]
```diff
@@ -56,6 +56,11 @@ public ArtifactEntity(
     var validPrefix = artifactStoreConfig.getPathPrefixWithSeparator() + this.entityName;
     var path = validPrefix + "/" + entityId + "/" + artifact.getKey();
 
+    var filenameExtension = artifact.getFilenameExtension();
+    if (!filenameExtension.isEmpty() && !filenameExtension.endsWith("." + filenameExtension)) {
```
Comment (liuverta) [inline backend/src/main/java/ai/verta/modeldb/entities/ArtifactEntity.java]:
```suggestion
    if (!filenameExtension.isEmpty() && !path.endsWith("." + filenameExtension)) {
```

**Review by liuverta — CHANGES_REQUESTED** [review]

File: backend/common/src/main/java/ai/verta/modeldb/common/subtypes/CommonArtifactHandler.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/common/src/main/java/ai/verta/modeldb/common/subtypes/CommonArtifactHandler.java]: Fixed

File: backend/src/main/java/ai/verta/modeldb/entities/ArtifactEntity.java (hunk 3 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/entities/ArtifactEntity.java]: Fixed

**Review by liuverta — APPROVED** [review]

**Pull request merged**